### PR TITLE
Allow to get a local cloud id without going through the contacts manager

### PR DESCRIPTION
**Pull request by juliusknorr**
As from https://github.com/nextcloud/activity/pull/617, [getCloudId can be expensive](https://github.com/nextcloud/server/blob/master/lib/private/Federation/CloudIdManager.php#L84-L120), especially on large instances since curently the contacts manager is called to search for the cloud id which may lead to slow queries as it searches the oc_card_properties table with a LIKE with wildcards on every file action:

```
SELECT DISTINCT `cp`.`cardid` FROM `oc_cards_properties` `cp` WHERE (`cp`.`addressbookid` = N) AND (`cp`.`name` = N) AND (`cp`.`value`  COLLATE utfNmbN_general_ci LIKE '%cloud@exmaple.com%');
```

For local users this PR allows to omit the remote parameter when getting the cloud id for a local user. The display name will then be fetched from the user manager instead and the remote URL will be constructed as it was done before in the User class.

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

**Comment by nickvergessen** [comment]
Should we do a generic short cut in the getCloudId() for the local server?

**Comment by juliushaertl** [comment]
@nickvergessen I added a shortcut as the id is also just constructed like this in the CloudIdManager, but having the possibility to request a local cloud id still makes sense for the manager I'd say, so I kept that.

File: lib/private/Federation/CloudIdManager.php [hunk 1]
```diff
@@ -103,24 +111,39 @@ protected function getDisplayNameFromContact(string $cloudId): ?string {
 
 	/**
 	 * @param string $user
-	 * @param string $remote
+	 * @param string|null $remote
```
Comment (blizzz) [inline lib/private/Federation/CloudIdManager.php:114]: null is not allowed in the parameter list

File: lib/public/Federation/ICloudIdManager.php [hunk 2]
```diff
@@ -46,12 +46,12 @@ public function resolveCloudId(string $cloudId): ICloudId;
 	 * Get the cloud id for a remote user
 	 *
 	 * @param string $user
-	 * @param string $remote
+	 * @param string|null $remote (optional since 23.0.0 for local users)
```
Comment (blizzz) [inline lib/public/Federation/ICloudIdManager.php:49]: TypeHint in parameter is not nullable

File: lib/private/Federation/CloudIdManager.php (hunk 1 above)
Comment (juliushaertl) [reply lib/private/Federation/CloudIdManager.php:114]: Made it nullable

**Review by blizzz — APPROVED** [review]

**Pull request merged**